hpobench/benchmarks/surrogates/paramnet_benchmark.py

```diff
@@ -0,0 +1,380 @@
+                                fidelity: Union[Dict, None] = None,
+                                rng: Union[np.random.RandomState, int, None] = None, **kwargs) -> Dict:
+        assert fidelity['step'] == 50, f'Only querying a result for the 50. epoch is allowed, ' \
+                                       f'but was {fidelity["step"]}.'
+        return self.objective_function(configuration, fidelity={'step': 50}, rng=rng)
+
+    @staticmethod
+    def get_fidelity_space(seed: Union[int, None] = None) -> CS.ConfigurationSpace:
+        """
+        Creates a ConfigSpace.ConfigurationSpace containing all fidelity parameters for
+        the SupportVector Benchmark
+
+        Fidelities
+        ----------
+        step: float - [1, 50]
+            Step, when to query the surrogate model
+
+        Parameters
+        ----------
+        seed : int, None
+            Fixing the seed for the ConfigSpace.ConfigurationSpace
+
+        Returns
+        -------
+        ConfigSpace.ConfigurationSpace
+        """
+        seed = seed if seed is not None else np.random.randint(1, 100000)
+        fidel_space = CS.ConfigurationSpace(seed=seed)
+
+        fidel_space.add_hyperparameters([
+            CS.UniformIntegerHyperparameter("step", lower=1, upper=50, default_value=50, log=False),
+        ])
+        return fidel_space
+
+    @staticmethod
+    def get_meta_information():
+        """ Returns the meta information for the benchmark """
+        meta_info = _ParamnetBase.get_meta_information()
+        meta_info.update({'info': 'This benchmark uses the epochs as fidelity.'})
+        return meta_info
+
+
+class _ParamnetOnTimeBenchmark(_ParamnetBase):
+
+    @AbstractBenchmark._configuration_as_dict
+    @AbstractBenchmark._check_configuration
+    @AbstractBenchmark._check_fidelity
+    def objective_function(self, configuration: Union[Dict, CS.Configuration],
+                           fidelity: Union[Dict, None] = None,
+                           rng: Union[np.random.RandomState, int, None] = None, **kwargs) -> Dict:
+
+        config_array = self.convert_config_to_array(configuration)
+        lc = self.surrogate_objective.predict(config_array)[0]
+        costs = self.surrogate_costs.predict(config_array)[0]
+
+        # If we can't afford a single epoch, return TODO.
+        if (costs / self.n_epochs) > fidelity['budget']:
+            # TODO: Return random performance here instead
+            y = 1
+            return {'function_value': float(y),
+                    'cost': fidelity['budget'],
+                    'info': {'state': 'Not enough budget'}}
+
+        learning_curves_cost = np.linspace(costs / self.n_epochs, costs, self.n_epochs)
+
+        if fidelity['budget'] < costs:
+            idx = np.where(learning_curves_cost < fidelity['budget'])[0][-1]
+            y = lc[idx]
+            lc = lc[:idx]
+
+        else:
+            # If the budget is larger than the actual runtime, we extrapolate the learning curve
```

Comment:
This has been done before, could you please add a line to the meta-info where you took the this from (I assume hpolib1.5)